### PR TITLE
[bridge] Extracting updating of prebuilds into PrebuildUpdater (2/3)

### PR DESCRIPTION
**Pull request by geropl**
## Description

This PR extract all logic related to updating prebuilds from src/ee/bridge behind a new interface `PrebuildUpdater` with two implementations: Noop (non-ee version) and DB (ee version) (2/3 in the chain)

## Related Issue(s)

Context: https://github.com/gitpod-io/gitpod/issues/9395

## How to test
 - start a workspace on this branch
 - start a workspace in the preview env
 - check the logs `kubectl logs ws-manager-bridge- ws-manager-bridge`
 - note how:
   - everything is working as expected
   - bridge logs show no errors

## Release Notes

```release-note
NONE
```

### CODE REVIEW

**Review by AlexTugarev — APPROVED** [review]
LGTM

/hold because of merge conflicts after 1/3 got merged

File: components/ws-manager-bridge/src/prebuild-updater.ts
Comment (laushinka) [inline components/ws-manager-bridge/src/prebuild-updater.ts:2]: Is this supposed to be 2020?

File: components/ws-manager-bridge/src/bridge.ts
Comment (svenefftinge) [inline components/ws-manager-bridge/src/bridge.ts:469]: @geropl I don't think this does anything useful. Looks to me like it would update the workspace instance that got set to 'stopped' a few lines above now to 'aborted'. Am I missing something?

Comment (geropl) [reply components/ws-manager-bridge/src/bridge.ts:469]: > Looks to me like it would update the workspace instance that got set to 'stopped' a few lines above now to 'aborted'. Am I missing something?

It updates the `PrebuiltWorkspace` that's corresponding that `WorkspaceInstance` to `aborted`, which would otherwise not bet updated.

In what context are you looking into this? Can I help?

Comment (svenefftinge) [reply components/ws-manager-bridge/src/bridge.ts:469]: > It updates the PrebuiltWorkspace that's corresponding that WorkspaceInstance to aborted, which would otherwise not bet updated.

Why does it do that? And how can a prebuild not be finished when there is a workspace based on it?

> In what context are you looking into this? Can I help?

Just reading the code and trying to make sense of it 😇

Comment (svenefftinge) [reply components/ws-manager-bridge/src/bridge.ts:469]: Ah, got it now. Forget it 😁

Comment (svenefftinge) [reply components/ws-manager-bridge/src/bridge.ts:469]: What confused me is that the method is called `stopPrebuildInstance` while it only aborts the PrebuildWorkspace in the database.

**Comment by geropl** [comment]
> Why does it do that? And how can a prebuild not be finished when there is a workspace based on it?

I think we went down the road of having an explicit state model for Prebuilds to better decouple tables and implementation.

> Just reading the code and trying to make sense of it 

:+1:

File: components/ws-manager-bridge/src/bridge.ts
Comment (geropl) [reply components/ws-manager-bridge/src/bridge.ts:469]: All good :wink:

**Pull request merged**
